driver/src/main/scala/api/commands/writeresult.scala

```diff
@@ -32,8 +32,7 @@ sealed trait WriteResult {
 object WriteResult {
   def lastError(result: WriteResult): Option[LastError] = result match {
```

Comment:
Even if there could be doubt about `WriteResult`, this case is documented and doesn't loose error detail.

On the other side, bulk is loosing such detail, and is the only case when there could be error even if `ok`.